### ISSUE TITLE
React Native 0.73 support

### ISSUE THREAD

**Issue by mozzius** [andrew-levy/react-native-safari-extension#24]
Hello. I'm unable to update to Expo 50 due to a problem with the React Native Podfile post-install hook.

```
[!] An error occurred while processing the post-install hook of the Podfile.

[!] The plist file at path `/Users/samuel.newman/Documents/Other/graysky/apps/expo/ios/Info.plist` doesn't exist.


/opt/homebrew/Cellar/cocoapods/1.14.3/libexec/gems/xcodeproj-1.23.0/lib/xcodeproj/plist.rb:17:in `read_from_path'
/Users/samuel.newman/Documents/Other/graysky/node_modules/react-native/scripts/cocoapods/utils.rb:557:in `block in update_ats_in_plist'
/Users/samuel.newman/Documents/Other/graysky/node_modules/react-native/scripts/cocoapods/utils.rb:554:in `each'
/Users/samuel.newman/Documents/Other/graysky/node_modules/react-native/scripts/cocoapods/utils.rb:554:in `update_ats_in_plist'
/Users/samuel.newman/Documents/Other/graysky/node_modules/react-native/scripts/cocoapods/utils.rb:579:in `apply_ats_config'
/Users/samuel.newman/Documents/Other/graysky/node_modules/react-native/scripts/react_native_pods.rb:310:in `react_native_post_install'
/Users/samuel.newman/Documents/Other/graysky/apps/expo/ios/Podfile:57:in `block (3 levels) in from_ruby'
```

Seems it tries to find all `Info.plist` files using this code:

```ruby
def self.get_plist_paths_from(user_project)
  info_plists = user_project
    .files
    .select { |p|
      p.name&.end_with?('Info.plist')
    }
  return info_plists
end
```

However, printing all the paths for these files shows this:

```
Expo.plist
Graysky/AppDelegate.h
Graysky/AppDelegate.mm
Graysky/Images.xcassets
Graysky/Info.plist
Graysky/main.m
Graysky/SplashScreen.storyboard
Graysky/noop-file.swift
Graysky/Graysky-Bridging-Header.h
Graysky.app
System/Library/Frameworks/JavaScriptCore.framework
libPods-Graysky.a
Target Support Files/Pods-Graysky/Pods-Graysky.debug.xcconfig
Target Support Files/Pods-Graysky/Pods-Graysky.release.xcconfig
Pods/Target Support Files/Pods-Graysky/ExpoModulesProvider.swift
src
assets
manifest.json
Info.plist
SafariWebExtensionHandler.swift
OpenInGrayskyExtension.appex
```

Since these extension files don't have the correct paths (specifically the `Info.plist` file), when it tries to open `/ios/Info.plist` it breaks.

**Comment by andrew-levy** [comment]
Thanks as always @mozzius! I'll take a look at this today. Looks like other config plugins are seeing the same thing: https://github.com/OneSignal/onesignal-expo-plugin/issues/213

**Comment by mozzius** [comment]
Great! I've patched `react-native` for now (not sure if it breaks the extension but at least it builds):

https://github.com/mozzius/graysky/blob/main/patches/react-native%400.73.1.patch

**Comment by andrew-levy** [comment]
@mozzius this is looking like a bug in React Native. There are a few issues and PRs related to this, check out Evans comment here https://github.com/facebook/react-native/pull/38086#issuecomment-1863490019

I'm gona hold off on a fix for this to see if this is fixed upstream. But if not i'll make a fix

**Comment by andrew-levy** [comment]
@mozzius This should be fixed in RN v0.73.3 once released https://github.com/OneSignal/onesignal-expo-plugin/issues/213#issuecomment-1914788799
